### PR TITLE
Add support for any language:

### PR DESCRIPTION
**Pull request by dmitry-timofeev**
Auto-detect the source language and the alias to use using the Pygments API.

### CODE REVIEW

**Comment by dmitry-timofeev** [comment]
It appears [`codehilite`](https://squidfunk.github.io/mkdocs-material/extensions/codehilite/) is used widely as an markdown extension, and it does not work with all _file_ extensions (e.g., "cs" does not yield good enough results for C# — must be "csharp").

I'll see if the correct info can be extracted using the Pygments API: https://pygments.org/

**Comment by rnorth** [comment]
Thanks for looking into this - it's a good idea. I'd be interested to see if there's a suitable source for the extension->language mapping.

**Comment by dmitry-timofeev** [comment]
It seems Pygments provides such a mapping through a pair of [`get_lexer_for_filename`](https://pygments.org/docs/api/#pygments.lexers.get_lexer_for_filename) and `aliases`. 
As Pygments is used by [codehilite](https://squidfunk.github.io/mkdocs-material/extensions/codehilite/), it must be the best source of such mapping. I updated the code to use it and added more tests.

File: tests/codeinclude/test_languages.py [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+class MyTestCase(unittest.TestCase):
+    def test_get_lang_class(self):
+        self.assertEquals('java', get_lang_class('HelloWorld.java'))
+        self.assertEquals('python', get_lang_class('HelloWorld.py'))
+        self.assertEquals('csharp', get_lang_class('HelloWorld.cs'))
+        self.assertEquals('rust', get_lang_class('HelloWorld.rs'))
+        self.assertEquals('docker', get_lang_class('Dockerfile'))
+        self.assertEquals('xml', get_lang_class('HelloWorld.xml'))
+        self.assertEquals('toml', get_lang_class('HelloWorld.toml'))
```
Comment (rnorth) [inline tests/codeinclude/test_languages.py:13]: Hmm, the test seems to fail on this line - I wonder if there's varying support for the `toml` extension in different versions of pygments.

```
>       self.assertEquals('toml', get_lang_class('HelloWorld.toml'))
E       AssertionError: 'toml' != 'none'
E       - toml
E       + none
```

**Review by rnorth — CHANGES_REQUESTED** [review]
Test failure when running this through `pytest`...

File: tests/codeinclude/test_languages.py (hunk 1 above)
Comment (dmitry-timofeev) [reply tests/codeinclude/test_languages.py:13]: It seems I used `Pygments~=2.5` or newer, which version fails for you? Would it make sense to require a certain version of it, or just remove this test?

**Comment by rnorth** [comment]
Looks like this is included in #15 
Trying to unpick these and forming a larger PR, #22, where I'll resolve the conflicts that have arisen

**Pull request closed without merging**